server/form-pages/apply/risks-and-needs/risk-of-serious-harm/oldOasys.test.ts

```diff
@@ -23,10 +32,25 @@ describe('OldOasys', () => {
         hasOldOasys: 'Confirm whether they have an older OASys with risk of serious harm (RoSH) information',
       })
     })
-    it('returns an error when hasOldOasys is yes but oasysCompletedDate is blank', () => {
-      const page = new OldOasys({ hasOldOasys: 'yes' }, application)
-      expect(page.errors()).toEqual({
-        oasysCompletedDate: 'Enter the date the OASys was completed',
+    describe('when hasOldOasys is yes', () => {
+      it('returns an error when oasysCompletedDate is blank', () => {
+        const page = new OldOasys({ hasOldOasys: 'yes' }, application)
+        expect(page.errors()).toEqual({
+          oasysCompletedDate: 'Enter the date the OASys was completed',
+        })
+      })
+
+      describe('when the date is not valid', () => {
+        beforeEach(() => {
+          jest.resetAllMocks()
+          ;(dateAndTimeInputsAreValidDates as jest.Mock).mockImplementation(() => false)
+        })
+        it('returns an error ', () => {
+          const page = new OldOasys({ hasOldOasys: 'yes' }, application)
```

Comment:
for this test to reach the `dateAndTimeInputsAreValidDates` function, does some value have to be given for `oasysCompletedDate` in the data blob?

Reply:
oh yes ofc. I'll mock that as well. gimme a sec